service/geopmdpy_test/TestActiveSessions.py

```diff
@@ -0,0 +1,648 @@
+        "signals" : ["ENERGY_DRAM", "FREQUENCY_MAX", "MSR::DRAM_ENERGY_STATUS:ENERGY"],
+        "controls" : ["CPU_FREQUENCY_CONTROL", "MSR::IA32_PERFEVTSEL0:CMASK"],
+        "watch_id" : 550
+    }
+    json_wrong_data_types = {
+        "client_pid" : "450",
+        "mode" : "r",
+        "signals" : ["ENERGY_DRAM", "FREQUENCY_MAX", "MSR::DRAM_ENERGY_STATUS:ENERGY"],
+        "controls" : ["CPU_FREQUENCY_CONTROL", "MSR::IA32_PERFEVTSEL0:CMASK"],
+        "watch_id" : "550"
+    }
+    json_additional_properties = {
+        "client_pid" : 450,
+        "mode" : "r",
+        "signals" : ["ENERGY_DRAM", "FREQUENCY_MAX", "MSR::DRAM_ENERGY_STATUS:ENERGY"],
+        "controls" : ["CPU_FREQUENCY_CONTROL", "MSR::IA32_PERFEVTSEL0:CMASK"],
+        "actuators" : ["ENERGY_DRAM", "FREQUENCY_MIN", "MSR::IA32_PERFEVTSEL0:CMASK"],
+        "watch_id" : 550,
+        "batch_id" : 450,
+    }
+    json_list_of_2 = [ json_good_example, json_good_example_2 ]
+    json_empty_dictionary = {}
+    string_empty_file = ""
+    string_typos_json = """{
+        "client_pid" : 450,
+        "mode" : "r"
+        "signals",  ["ENERGY_DRAM", "FREQUENCY_MAX", "MSR::DRAM_ENERGY_STATUS:ENERGY"],
+        "controls" : ["CPU_FREQUENCY_CONTROL", "MSR::IA32_PERFEVTSEL0:CMASK"],
+        "watch_id" 550,
+    }
+    """
+    string_c_code = """
+    #include <stdio.h>
+    #include <stdlib.h>
+    int main(void) {
+        int i = 0;
+        for (i = 0; "Hello World"[i] != '\0'; i++)
+            putchar(i["Hello World"]);
+        return i;
+    }
+    """
+
+    def setUp(self):
+        """Create temporary directory
+
+        """
+        self._test_name = 'TestActiveSessions'
+        self._TEMP_DIR = tempfile.TemporaryDirectory(self._test_name)
+
+    def tearDown(self):
+        """Clean up temporary directory
+
+        """
+        self._TEMP_DIR.cleanup()
+
+    def check_dir_perms(self, path):
+        """Assert that the path points to a file with mode 0o700
+        Assert that the user and group have the right permissions.
+
+        """
+        st = os.stat(path)
+        perm_mode = stat.S_IMODE(st.st_mode)
+        user_owner = st.st_uid
+        group_owner = st.st_gid
+        self.assertEqual(0o700, perm_mode)
+        self.assertEqual(os.getuid(), user_owner)
+        self.assertEqual(os.getgid(), group_owner)
+
+    def check_getters(self, session, client_pid, signals, controls, watch_id):
+        self.assertIn(client_pid, session.get_clients())
+        self.assertEqual(signals, session.get_signals(client_pid))
+        self.assertEqual(controls, session.get_controls(client_pid))
+        self.assertEqual(watch_id, session.get_watch_id(client_pid))
+        self.assertFalse(session.get_batch_server(client_pid))
+        self.assertFalse(session.is_write_client(client_pid))
+
+    def create_json_file(self, directory, filename, contents, permissions=0o600):
+        """Create a json file that  matches the ActiveSessions._session_schema
```

Comment:
This helper does create a JSON file, but it does not validate it against the session schema, so there are no guarantees that it matches the schema right now.

Was that intended?